vllm/executor/ray_gpu_executor.py

```diff
@@ -23,12 +22,28 @@
 
 logger = init_logger(__name__)
 
+# If the env var is set, it uses the Ray's compiled DAG API
+# which optimizes the control plane overhead.
+# Run vLLM with VLLM_USE_RAY_COMPILED_DAG=1 to enable it.
+# Currently, this requires USE_SPMD_WORKER=True.
 USE_RAY_COMPILED_DAG = envs.VLLM_USE_RAY_COMPILED_DAG
+# If the env var is set, then we do not distinguish between the "driver worker"
+# vs other workers. Also, the rank 0 worker will be executed in a remote Ray
+# worker. Currently this requires USE_RAY_COMPILED_DAG=True.
+USE_SPMD_WORKER = envs.VLLM_USE_SPMD_WORKER
```

Comment:
read these environment variable inside `_init_executor`, so that they are lazily read, and you can change them for multiple times during test.

Reply:
Done